src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -696,9 +696,17 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     handleOne(callbacks, idx + 1);
                 } else if (callback instanceof AuthorizeCallback) {
                     final AuthorizeCallback authorizeCallback = (AuthorizeCallback) callback;
+                    String authenticationID = authorizeCallback.getAuthenticationID();
+                    if (authenticationID == null) {
+                        authenticationID = getAuthenticationPrincipal().getName();
```

Comment:
I've tried to keep the same behavior and just assign the authcID from the authenticated principal. I also think it is redundant, will remove that and do what you suggest.